dialogue-annotations/src/main/java/com/palantir/dialogue/annotations/RequestDeserializer.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.annotations;
+
+import com.palantir.dialogue.Deserializer;
+import com.palantir.dialogue.Response;
+import java.util.Optional;
+
+public enum RequestDeserializer implements Deserializer<Response> {
+    INSTANCE;
+
+    @Override
+    public Response deserialize(Response response) {
+        return response;
+    }
+
+    @Override
+    public Optional<String> accepts() {
+        return Optional.empty();
```

Comment:
Perhaps something along these lines?
```suggestion
        return Optional.of("*/*");
```

Reply:
Sorry clicked commit too quickly. This is an interesting question I believe: I can imagine that maybe there might be some usecase where you'd like to be able to affect this? And therefore should perhaps be not a special thing here, but you need to implement the deserializer yourself?

Reply:
Actually, this is perhaps an OK default, but we should always allow the escape hatch of:

What I mean is:

```
    @Request(method = HttpMethod.PUT, path = "/custom/request1", accept = MyCustomResponseDeserializer.class)
    Response customResponse();
```

Reply:
For what it's worth, I think we could leave this deserializer out for the time being, there's some additional ambiguity around whether users will expect it to succeed for non-200 responses, so it may be best to provide developers with the tools to do the right thing. In most cases I think it's preferable if they return a custom type instead of the Response object anyhow.

Reply:
To be clear, I don't have a strong opinion on this and defer to your judgement. Only pointing out how a standard implementation might turn out to be a foot-gun.

Reply:
That's exactly what I meant: this can be covered by a custom deserializer that anyone can implement, so we should take this out. It was in the example on the Quip, so I impl'ed it, but it doesn't really make sense. Or rather, I'll have it as a custom one in the example, just to show it off, but it won't be in the -annotations project.

Reply:
Alright, I moved it around and added some docs to Json class to point out that it's conjure style json, including special error behavior.